### PR TITLE
Enable adding analytics to template ad

### PR DESCRIPTION
**Pull request by zhouyx**
Insert `<amp-analytics>` component to template ad.

creativeJson should be defined like
```
{
  "templateUrl": ...
  "data": {}
  "analytics": {
     "remote": // define remote config url
     "type": // type attribute
     "inline": // inline config valeu
   } // Can also be an array of jsonObject
}
```

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-ad-templates.js [hunk 1]
```diff
@@ -75,7 +78,39 @@ export class AmpAdTemplates {
    */
   render(templateValues, element) {
     return Services.templatesFor(this.win_)
-        .findAndRenderTemplate(element, templateValues);
+        .findAndRenderTemplate(element, templateValues)
+        .then(renderedElement => {
```
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-ad-templates.js]: is the `then` clause needed?

File: extensions/amp-a4a/0.1/amp-ad-templates.js [hunk 2]
```diff
@@ -75,7 +78,39 @@ export class AmpAdTemplates {
...
+  /**
+   * @param {!Element} element
+   * @param {!Array|!JsonObject} analyticsValue
+   */
+  insertAnalytics(element, analyticsValue) {
+    analyticsValue = /**@type {!Array}*/
+        (isArray(analyticsValue) ? analyticsValue : [analyticsValue]);
+    for (let i = 0; i < analyticsValue.length; i++) {
+      const config = analyticsValue[i];
+      const analyticsEle = document.createElement('amp-analytics');
```
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-ad-templates.js]: should you use `element.owenerDocument`

File: extensions/amp-a4a/0.1/amp-ad-templates.js [hunk 3]
```diff
@@ -75,7 +78,39 @@ export class AmpAdTemplates {
...
+      if (config['remote']) {
+        analyticsEle.setAttribute('config', config['remote']);
+      }
+      if (config['type']) {
+        analyticsEle.setAttribute('type', config['type']);
+      }
+      if (config['inline']) {
+        const scriptElem = createElementWithAttributes(
+            document,
```
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-ad-templates.js]: same here

File: extensions/amp-ad-network-adzerk-impl/0.1/amp-ad-network-adzerk-impl.js [hunk 4]
```diff
@@ -151,6 +151,18 @@ export class AmpAdNetworkAdzerkImpl extends AmpA4A {
 
   /** @override */
   getAmpAdMetadata(unusedCreative) {
+    if (this.ampCreativeJson_.analytics) {
```
Comment (lannka) [inline extensions/amp-ad-network-adzerk-impl/0.1/amp-ad-network-adzerk-impl.js]: should you just load the analytics extension?

File: extensions/amp-a4a/0.1/amp-ad-templates.js (hunk 1 above)
Comment (zhouyx) [reply extensions/amp-a4a/0.1/amp-ad-templates.js]: removed.

File: extensions/amp-a4a/0.1/amp-ad-templates.js (hunk 2 above)
Comment (zhouyx) [reply extensions/amp-a4a/0.1/amp-ad-templates.js]: fixed.

File: extensions/amp-a4a/0.1/amp-ad-templates.js (hunk 3 above)
Comment (zhouyx) [reply extensions/amp-a4a/0.1/amp-ad-templates.js]: fixed

File: extensions/amp-ad-network-adzerk-impl/0.1/amp-ad-network-adzerk-impl.js (hunk 4 above)
Comment (zhouyx) [reply extensions/amp-ad-network-adzerk-impl/0.1/amp-ad-network-adzerk-impl.js]: I prefer not to load the extension here. AmpA4A will load extension based on metadata, I think it's better to load all extensions together.

Comment (zhouyx) [reply extensions/amp-ad-network-adzerk-impl/0.1/amp-ad-network-adzerk-impl.js]: Talked offline. It's not that straightforward to load extension here. Will keep the current approach.

**Pull request merged**